web_services/foxx/brightid/db.js

```diff
@@ -948,62 +660,109 @@ function removeSigningKey(id, signingKey) {
   usersColl.update(id, { signingKeys });
 }
 
-function removeAllSigningKeys(id, signingKey) {
-  let signingKeys = usersColl.document(id).signingKeys || [];
+function removeAllSigningKeys(userId, signingKey) {
+  let signingKeys = usersColl.document(userId).signingKeys || [];
   signingKeys = signingKeys.filter(s => s == signingKey);
-  usersColl.update(id, { signingKeys });
+  usersColl.update(userId, { signingKeys });
+}
+
+function vouchFamily(userId, groupId, timestamp) {
+  const group = getGroup(groupId);
+  if (group.type != 'family') {
+    throw new errors.NotFamilyGroupError();
+  }
+
+  // users can start vouching only after all invitees join
+  if (groupInvites(groupId).length > 0 || groupMembers(groupId).length < 2) {
+    throw new errors.IneligibleToVouch();
+  }
+
+  const members = groupMembers(groupId);
+  const conns = knownConnections(userId);
+  // if some of group members are unknown to voucher
+  if (_.intersection(members, conns).length != members.length) {
+    throw new errors.IneligibleToVouchFor();
+  }
+
+  group.vouchers.push(userId);
+  groupsColl.update(group, { vouchers: group.vouchers });
+}
+
+function userFamiliesToVouch(userId) {
+  checkUserExists(userId);
+  const result = [];
+  const conns = knownConnections(userId);
+  const connIds = conns.map(key => `users/${key}`);
+  const candidates = query`
+    FOR conn in ${usersInGroupsColl}
+      FILTER conn._from IN ${connIds}
+      FOR group in ${groupsColl}
+        FILTER group._id == conn._to
+        FILTER group.type == 'family'
+      RETURN DISTINCT group
+  `.toArray();
+  return candidates.filter(group => {
+    const members = groupMembers(group._key);
+    return !group.vouchers.includes(userId) &&
+      members.length >= 2 &&
+      groupInvites(group._key).length == 0 &&
+      _.intersection(members, conns).length == members.length
+  }).map(group => group._key);
+}
+
+function changeFamilyHead(admin, head, groupId) {
```

Comment:
This function needs to add a check that the existing family head is not a regular family member of another family group, i.e.

`! familyGroup(existingHead)`

This group will become the family group for the old head--assuming there is a head.

Reply:
Done